pkg/volume/empty_dir/empty_dir.go

```diff
@@ -281,19 +287,44 @@ func (ed *emptyDir) setupHugepages(dir string) error {
 		return nil
 	}
 
-	// TODO: get size and pageSize mount options from pods resources request
-	options, err := ed.getHugePagesMountOptions()
+	pageSizeMountOption, err := getPageSizeMountOptionFromPod(ed.pod)
 	if err != nil {
 		return err
 	}
 
 	glog.V(3).Infof("pod %v: mounting hugepages for volume %v", ed.pod.UID, ed.volName)
-	return ed.mounter.Mount("nodev", dir, "hugetlbfs", options)
-}
+	return ed.mounter.Mount("nodev", dir, "hugetlbfs", []string{pageSizeMountOption})
+}
+
+// getPageSizeMountOptionFromPod retrieves pageSize mount option from Pod's resources
+// and validates pageSize options in all containers of given Pod.
+func getPageSizeMountOptionFromPod(pod *v1.Pod) (string, error) {
+	pageSizeFound := false
+	pageSize := resource.Quantity{}
+	for _, container := range pod.Spec.Containers {
```

Comment:
its possible (but unlikely) only an init container would consume a hugepage resource, so check that list as well.